### ISSUE TITLE
[Request] Add setting for sentence continuation style

### ISSUE THREAD

**Issue by Flitskikker** [SubtitleEdit/subtitleedit#3417]
Similar to the [dialogue style request](https://github.com/SubtitleEdit/subtitleedit/issues/3416), the sentence continuation style is another subtitling element that's different across countries. At the moment, Subtitle Edit only supports one continuation style. I think it would be useful to allow the user to configure the continuation style used.

As far as I can tell, these are the 4 most common continuation styles:

_English / Flemish style (currently the only supported style):_
```
Lorem ipsum dolor sit amet
consectetur adipiscing elit,

donec eget turpis consequat
turpis commodo hendrerit

mauris mollis consectetur nibh,
nec congue est viverra quis.
```

_Dutch TV style:_
```
Lorem ipsum dolor sit amet
consectetur adipiscing elit...

donec eget turpis consequat
turpis commodo hendrerit...

mauris mollis consectetur nibh,
nec congue est viverra quis.
```

_Dutch cinema style:_
```
Lorem ipsum dolor sit amet
consectetur adipiscing elit...

...donec eget turpis consequat
turpis commodo hendrerit...

...mauris mollis consectetur nibh,
nec congue est viverra quis.
```

_Scandinavian style:_
```
Lorem ipsum dolor sit amet
consectetur adipiscing elit -

- donec eget turpis consequat
turpis commodo hendrerit -

- mauris mollis consectetur nibh,
nec congue est viverra quis.
```

As with the dialogue styles, there exist slight variations, for some styles also based on the amount of time between the subtitles, so I think it would be best to also implement generic settings, for example:

![se4](https://user-images.githubusercontent.com/3516155/53379666-c838e800-396a-11e9-96cf-61bc93ce5b29.png)

This way, users can configure it completely.

These settings would then be used across the whole application, for example:
- When splitting a subtitle (suffix and prefix get added)
- When merging two subtitles (suffix and prefix get removed)
- For detecting errors (e.g. missing prefix/suffix)
- For the "Only merge continuation lines" option in Tools > Merge short lines
- If possible, an option to convert between styles

I was working on a plugin myself to automatically convert from the English continuation style to the other continuation styles. Unfortunately, the development got stranded because there was just too much to take into account. Detecting new sentences, detecting lyrics and translated signs, adding suffix/prefix inside tags, adding suffix/prefix inside dialogues... Maybe you could have a try at it, then it would also be integrated into SE directly. I can send you thoughts / things to take into account separately.

This would really speed up some things when I'm working with SE. It would be a miracle to be able to add continuation dots to subtitles that don't have them in one click instead of fiddling with regex in Notepad++, since I couldn't get it to work.

Again, thanks in advance! :)

Information on why continuation dots are recommended:
* http://www.bokorlang.com/journal/04stndrd.htm (see the first two points under "4. Punctuation and letter case")

**Comment by Ding-adong** [comment]
Why would you add ... in the beginning of new line? All it does is take 3 chars, thus 3 less chars for actual letters, for no reason. It would mean that when there is a lot of talking, timing becomes restricted and more lines used.
A new line with LC means the same speaker as previous line. New speaker would have new line with UP.
Ending a line with ... and the next line with LC is the same speaker.
Ending a line with ... and the next line with UP is a new speaker interrupting the previous speaker before finishing one's sentence.

English style use , if the sentence requires it. If the sentence doesn't require , then ... is used.
1:
I am fine.
2:
I am...
fine.
3:
I am...
Hi. (new speaker) on next new line.
4:
- I am...
- Hi.

On a personal note I use ... on the end of the line as in example 3, 4. Not for example 2.
5:
I am
fine.
Since there is no full stop . I know the it is the same speaker because common sense tells me so. To me ... in example 2 is pointless since no one use it when writing a book/letter/report etc. Why use it in subtitles?

**Comment by Ding-adong** [comment]
http://www.bokorlang.com/journal/04stndrd.htm is an idiot. Most of his proposals are pointless and irrelevant.

**Comment by Flitskikker** [comment]
Thank you for your insights.

I also agree that adding ... to the beginning of a continuing sentence is pointless. However, they are used in Dutch cinema subtitles and in Dutch Netflix translations, so it would be nice to be able to convert existing subtitles for Netflix/cinema usage.
https://partnerhelp.netflixstudios.com/hc/en-us/articles/215350158-Dutch-Timed-Text-Style-Guide

Adding ... to the end of continuing sentences is actually very important, as it provides the viewer with a clear indication that the sentence is not finished. I also experience it myself as a viewer; the missing dot is easy to miss, so when the first title is a good sentence as-is, my brain interprets it as a completed sentence. When, in the next title, the sentence suddenly continues, it is very confusing.

Something similar happens when a continuing sentence ends with a comma, which can be misread as a full stop. When, in the next title, the sentence suddenly continues, it is very confusing as well.

The thing with subtitles is that the viewer only has one chance to read the text. Therefore, subtitles should be as clear and as less confusing as possible. Using connection dots (as suffix only, then) provides the viewer with clarity.

This practice has been used since the beginning of Dutch and Scandinavian subtitling.

Furthermore, if you spot '...' in a subtitle where connection dots are not used, you **know** that the speaker doesn't finish their sentence, possibly ruining an element of surprise, because the viewer will anticipate on that (for example, they get suddenly killed mid-sentence).

> http://www.bokorlang.com/journal/04stndrd.htm is an idiot. Most of his proposals are pointless and irrelevant.

I also don't agree with a lot of his proposals, but he had a good explanation on why connection dots are recommended.

**Comment by Ding-adong** [comment]
Getting killed mid-sentence is a rare thing ha ha. The problem is hearing people lack the same experience as deaf people in 'reading' subtitles. Deaf people don't 'read' subtitles, we scan them in a flash (no different from blinking) and watch the programme. Reading subtitles takes too long and thus less time watching the programme and missing out on the fun. Does it matter if you fail to spot a , or . ? If you scan the subtitles and watch the programme, you will understand what going on by watching the characters. Clarity comes from watching not reading.

**Comment by Flitskikker** [comment]
I understand what you say, but when people **do** read the subtitles (for example if they don't understand the language spoken), the subtitles should read flawlessly and not be potentially confusing.

To ensure that the viewer can both follow the fun **and** 'read' the subtitles, we use a (relatively low) reading speed of around 11 CPS (with an upper limit of 15 CPS). This is proven to be an optimal reading speed that doesn't require the viewer to be actively reading subtitles (i.e. they don't notice that they're reading the subtitles). Anything above 14/15 CPS (basically every other language exceeds this limit) **will** require the viewer to actively read subtitles.

Anyhow, the connection dots (or dashes) are an established convention in Dutch and Scandinavian subtitling, not something I came up with and only use myself. Therefore, it would be useful to support configurable connection styles for people who subtitle into these languages.

**Comment by taxen** [comment]
So... how thing going on this matter?
I too have interest with adding (at least) another continuation style, especially for "Only merge continuation lines" option in Tools > Merge short lines.
Because the current style disregards a dialog style and takes it as a continuation, merging subscripts with a dialog makes one subscript with a dialog in it. As I pointed out earlier https://github.com/SubtitleEdit/subtitleedit/issues/3527#issuecomment-548344293 , there should be a way to exclude a dialog when merging.

**Comment by Flitskikker** [comment]
Fixed by #4108 / 9645ee156cf74a5b27d1efd9e82397576a79d4eb.
